lib/detect-testing-library-utils.ts

```diff
@@ -335,12 +335,20 @@ export function detectTestingLibraryUtils<
      * Not to be confused with {@link isUserEventMethod}
      */
     const isUserEventUtil = (node: TSESTree.Identifier): boolean => {
-      return isTestingLibraryUtil(
-        node,
-        (identifierNodeName, originalNodeName) => {
-          return [identifierNodeName, originalNodeName].includes('userEvent');
-        }
-      );
+      const userEvent = findImportedUserEventSpecifier();
+      let userEventName: string | undefined;
+
+      if (userEvent) {
+        userEventName = userEvent.name;
+      } else if (isAggressiveModuleReportingEnabled()) {
+        userEventName = USER_EVENT_NAME;
+      }
+
+      if (!userEventName) {
+        return false;
+      }
```

Comment:
Same issue I had with `isUserEventMethod`: `userEvent` util won't be imported from Testing Library module and it's a default import, so it has to be checked against its own import specifier.